Scripts/modelsystem.py

```diff
@@ -373,49 +346,76 @@ def _calculate_noise_areas(self):
+                self.resultdata.print_data(logsum, f"accessibility.txt")
+    
+    def _export_resultdata(self):
+        # Export purpose demand
+        gen_tours_purpose = {purpose.name: self._get_purpose_tours(purpose, attraction=False) 
+                             for purpose in self.dm.tour_purposes}
+        attr_tours_purpose = {purpose.name: self._get_purpose_tours(purpose, generation=False)
+                              for purpose in self.dm.tour_purposes}
+        self.resultdata.print_data(gen_tours_purpose, "zone_generation_by_purpose.txt")
+        self.resultdata.print_data(attr_tours_purpose, "zone_attraction_by_purpose.txt")
+
+        # Export mode demand
+        gen_tours_mode = {mode: self._get_mode_tours(mode, attraction=False)
+                           for mode in self.travel_modes}
+        attr_tours_mode = {mode: self._get_mode_tours(mode, generation=False)
+                           for mode in self.travel_modes}
+        self.resultdata.print_data(gen_tours_mode, "zone_generation_by_mode.txt")
+        self.resultdata.print_data(attr_tours_mode, "zone_attraction_by_mode.txt")
+
+        # Export mode shares by purpose
+        for purpose in self.dm.tour_purposes:
+            demsums = {mode: purpose.generated_tours[mode].sum() for mode in purpose.modes}
+            demand_all = float(sum(demsums.values()))
+            mode_shares = pandas.concat({purpose.name: pandas.Series(
+                {mode: demsums[mode] / demand_all for mode in demsums}, 
+                name="mode_share")}, names=["purpose", "mode"])
+            self.resultdata.print_concat(mode_shares, "purpose_mode_shares.txt")
+        
+        # Export tour length histograms
+        for purpose in self.dm.tour_purposes:
+            self.resultdata.print_concat(
+                pandas.concat(
+                    {m: purpose.histograms[m].histogram for m in purpose.histograms},
+                    names=["mode", "purpose", "interval"]),
+                "tour_lengths.txt")
+            self.resultdata.print_matrices(
+                purpose.aggregates, "aggregated_demand", purpose.name)
+        
+        # Export within zone demand
+        for purpose in self.dm.tour_purposes:
+            for mode in purpose.aggregates:
                 self.resultdata.print_data(
-                    pandas.Series(savu, zone_numbers, name="savu_zone"),
-                    "savu.txt")
-                avg_savu = numpy.searchsorted(intervals, avg_access) + 1
-                avg_savu += ((avg_access - intervals[avg_savu-2])
-                            / (intervals[avg_savu-1] - intervals[avg_savu-2]))
-                self.resultdata.print_line(
-                    f"Average SAVU:\t{avg_savu:1.4f}", "result_summary")
-
-    def _sum_trips_per_zone(self, mode, include_dests=True):
+                    purpose.within_zone_tours[mode], "within_zone_tours.txt")
+
+    def _get_mode_tours(self, mode, generation = True, attraction = True):
```

Comment:
This is just a small remark, but if both are `True` (the default), does it not return the number of *trips* (departing+arriving)?

Reply:
Yep, agreed. Didn't come with any good ideas of how implement this function (both generation and attraction). Probably having them separately as purpose methods is better.